File: python/Longest_Balanced_Subarray_I_3719.py
class Solution:
    def longestBalanced(self, nums: List[int]) -> int:
        maxWindow = 0
        for i in range(len(nums)):
            odd, even = set(), set()
            for j in range(i, len(nums)):
                if nums[j] & 1:
                    odd.add(nums[j])
                else:
                    even.add(nums[j])
                if len(odd) == len(even):
                    maxWindow = max(maxWindow, j - i + 1)
        return maxWindow

File: cpp/Longest_Balanced_Subarray_I_3719.cpp
class Solution {
public:
    int longestBalanced(vector<int>& nums) {
        int maxWindow = 0;
        for (int i = 0; i < nums.size(); i++) {
            set <int> odd, even;
            for (int j = i; j < nums.size(); j++) {
                if (nums[j] & 1) odd.insert(nums[j]);
                else even.insert(nums[j]);
                if (odd.size() == even.size()) maxWindow = max(maxWindow, j -i + 1);
            }
        }
        return maxWindow;
    }
};
